Assets/SEE/GraphProviders/GitGraphProvider.cs

```diff
@@ -0,0 +1,282 @@
+        /// <param name="oldRevision">the older revision that constitutes the baseline of the comparison</param>
+        /// <param name="newRevision">the newer revision against which the <see cref="oldRevision"/> is to be compared</param>
+        protected void AddLineofCodeChurnMetric(Graph graph, string vcsPath, string oldRevision, string newRevision)
+        {
+            using (var repo = new Repository(vcsPath))
+            {
+                var oldCommit = repo.Lookup<Commit>(oldRevision);
+                var newCommit = repo.Lookup<Commit>(newRevision);
+
+                var changes = repo.Diff.Compare<Patch>(oldCommit.Tree, newCommit.Tree);
+
+                foreach (var change in changes)
+                {
+                    foreach (var node in graph.Nodes())
+                    {
+                        if (node.ID.Replace('\\', '/') == change.Path)
+                        {
+                            node.SetInt("Lines Added", change.LinesAdded);
+                            node.SetInt("Lines Deleted", change.LinesDeleted);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Calculates the number of unique developers who contributed to each file for each file changed between two commits and adds it as a metric to <paramref name="graph"/>.
+        /// </summary>
+        /// <param name="graph">an existing graph where to add the metrics</param>
+        /// <param name="vcsPath">the path to the VCS containing the two revisions to be compared</param>
+        /// <param name="oldRevision">the older revision that constitutes the baseline of the comparison</param>
+        /// <param name="newRevision">the newer revision against which the <see cref="oldRevision"/> is to be compared</param>
+        protected void AddNumberofDevelopersMetric(Graph graph, string vcsPath, string oldRevision, string newRevision)
+        {
+            using (var repo = new Repository(vcsPath))
+            {
+                var oldCommit = repo.Lookup<Commit>(oldRevision);
+                var newCommit = repo.Lookup<Commit>(newRevision);
+
+                var commits = repo.Commits.QueryBy(new CommitFilter { SortBy = CommitSortStrategies.Topological });
+
+                var uniqueContributorsPerFile = new Dictionary<string, HashSet<string>>();
+
+                foreach (var commit in commits)
+                {
+                    if (commit.Author.When >= oldCommit.Author.When && commit.Author.When <= newCommit.Author.When)
+                    {
+                        foreach (var parent in commit.Parents)
+                        {
+                            var changes = repo.Diff.Compare<Patch>(parent.Tree, commit.Tree);
+
+                            foreach (var change in changes)
+                            {
+                                var filePath = change.Path;
+                                var id = commit.Author.Email;
```

Comment:
[Do not use `var`.](https://github.com/uni-bremen-agst/SEE/wiki/ReviewChecklist)